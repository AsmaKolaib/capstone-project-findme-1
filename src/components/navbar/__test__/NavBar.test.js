import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from './../NavBar';

import { render, cleanup } from '@testing-library/react';
// import 'jest-dom/extend-expect';

import renderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Navbar></Navbar>, div);
});

// it('renders button correctly', () => {
//   const { getByTestId } = render(<Navbar></Navbar>);
//   expect(getByTestId('navbar')).toHaveTextContent('');
// });

it('matches snapshot', () => {
  const tree = renderer.create(<Navbar></Navbar>).toJSON();
  expect(tree).toMatchSnapshot();
});
