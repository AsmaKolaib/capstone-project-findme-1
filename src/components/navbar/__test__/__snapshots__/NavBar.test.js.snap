// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<nav
  className="relative flex flex-wrap items-center navbar-expand-lg bg-tail-300 py-3 px-8 shadow"
  data-testid="navbar"
>
  <div
    className="w-full sm:w-full xl:w-48 px-4 mx-auto flex flex-wrap items-center justify-between"
  >
    <div
      className="w-full relative flex justify-between xl:w-auto xl:static xl:block xl:justify-start"
    >
      <a
        className="p-2 mr-4 inline-flex items-center"
        href="#"
      >
        <img
          alt="find me logo"
          className="h-8 w-8"
          src="FindMeLogo.svg"
        />
        <span
          className="text-xl text-black font-bold uppercase tracking-wide"
        >
          Find Me
        </span>
      </a>
      <button
        className="text-black inline-flex p-3 rounded xl:hidden ml-auto hover:text-white focus:outline-none outline-none nav-toggler items-center"
        data-target="#navigation"
        onClick={[Function]}
      >
        <img
          alt="menu icon"
          src="icon-menu.svg"
        />
      </button>
    </div>
  </div>
  <div
    className="hidden xl:flex flex-grow items-center"
  >
    <ul
      className="flex flex-col xl:flex-row list-none xl:ml-auto"
    >
      <li
        className="mt-2 xl:mt-0"
      >
        <a
          className="xl:inline-flex xl:w-auto w-full px-3 py-1 rounded-full text-black items-center justify-center hover:text-blue-600"
          href="#"
        >
          <span>
            Home
          </span>
        </a>
      </li>
      <li
        className="mt-2 xl:mt-0"
      >
        <a
          className="xl:inline-flex xl:w-auto w-full px-3 py-1 rounded-full text-black items-center justify-center hover:text-blue-600"
          href="#"
        >
          <span>
            Missing People
          </span>
        </a>
      </li>
      <li
        className="mt-2 xl:mt-0"
      >
        <a
          className="xl:inline-flex xl:w-auto w-full px-3 py-1 rounded-full text-black items-center justify-center hover:text-blue-600"
          href="#"
        >
          <span>
            About Us
          </span>
        </a>
      </li>
      <li
        className="mt-2 xl:mt-0"
      >
        <a
          className="xl:inline-flex xl:w-auto w-full px-3 py-1 rounded-full text-black items-center justify-center hover:text-blue-600"
          href="#"
        >
          <span>
            Contact Us
          </span>
        </a>
      </li>
      <li
        className="mt-2 xl:mt-0"
      >
        <div
          className="relative inline-block text-left"
        >
          <div>
            <span
              className="rounded-md shadow-sm"
            >
              <button
                aria-expanded="true"
                aria-haspopup="true"
                className="inline-flex justify-center w-full rounded-md px-4 py-2 bg-white text-sm leading-5 font-medium text-gray-700 hover:text-gray-500 focus:outline-none focus:border-blue-300 focus:shadow-outline-blue active:bg-gray-50 active:text-gray-800 transition ease-in-out duration-150"
                id="options-menu"
                onClick={[Function]}
                type="button"
              >
                <img
                  alt="language icon"
                  src="icon-language.svg"
                />
                <svg
                  className="-mr-1 ml-2 h-5 w-5"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                  />
                </svg>
              </button>
            </span>
          </div>
          <div
            className="hidden
          origin-top-right absolute left-0 mt-2 w-20 rounded-md shadow-lg"
          >
            <div
              className="rounded-md bg-white shadow-xs"
            >
              <div
                aria-labelledby="options-menu"
                aria-orientation="vertical"
                className="py-1"
                role="menu"
              >
                <a
                  className="block px-4 py-2 text-sm leading-5 text-gray-700 hover:bg-gray-100 hover:text-gray-900 focus:outline-none focus:bg-gray-100 focus:text-gray-900"
                  href="#"
                  role="menuitem"
                >
                  English
                </a>
                <a
                  className="block px-4 py-2 text-sm leading-5 text-gray-700 text-right hover:bg-gray-100 hover:text-gray-900 focus:outline-none focus:bg-gray-100 focus:text-gray-900"
                  href="#"
                  role="menuitem"
                >
                  عربي
                </a>
              </div>
            </div>
          </div>
        </div>
      </li>
      <li
        className="mt-2 xl:mt-0"
      >
        <a
          className=""
          href="#"
        >
          <button
            className="inline-flex xl:w-auto w-full mx-3 px-8 py-1 rounded-full text-gray-400 items-center justify-center border-2 h-8 border-blue-500 text-blue-500 focus:outline-none"
          >
            <img
              alt="search icon"
              className="pr-5 h-5"
              src="icon-search.svg"
            />
            Search
          </button>
        </a>
      </li>
      <li
        className="mt-2 xl:mt-0"
      >
        <a
          className=""
          href="#"
        >
          <button
            className="inline-flex xl:w-auto w-full mx-3 px-8 py-1 rounded-full text-white bg-gradient-to-l from-blue-700 to-blue-400 items-center justify-center focus:outline-none"
          >
            <img
              alt="post icon"
              className="pr-5 h-5"
              src="icon-post.svg"
            />
            Post
          </button>
        </a>
      </li>
      <li
        className="mt-2 xl:mt-0"
      >
        <a
          className=""
          href="www.google.com"
        >
          <button
            className="inline-flex xl:w-auto w-full mx-3 px-3 py-1 rounded-full text-blue-600 items-center justify-center focus:outline-none"
          >
            <img
              alt="google icon"
              className="pr-5 h-5"
              src="icon-SignInWithGoogle.png"
            />
            Sign in with Google
          </button>
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
